src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectLaunchTargetsProvider.cs

```diff
@@ -606,6 +608,11 @@ private static async Task<Guid> GetDebuggingEngineAsync(ConfiguredProject config
             return null;
         }
 
+        internal static string StripCommandLineArgs(string commandLineArgs)
+        {
+            return commandLineArgs.Replace(Environment.NewLine, " ");
```

Comment:
Are we sure that incoming new lines will always be `\r\n` (which would be the value of `Environment.NewLine` on Windows)? If `launchSettings.json` is saved with different line endings, I'm not sure they'd be normalised before arriving here.

One option would be:

```c#
return Regex.Replace(commandLineArgs, "[\r\n]+", " ")
```

Or in English: Replace any number of consecutive `\r` and/or `\n` characters with a single space.

Reply:
Is there any use of \r in modern Windows besides as part of a newline? If so, a safer regex would be (\n|\r\n)+

Reply:
> This looks good. You'll want to do a lot of testing here. As mentioned in comments, what happens if `launchSettings.json` is converted to other line endings (as might happen on a git checkout, for example)? What the value contains multiple newlines between values.
> 
> Consider writing a small console application that prints out its arguments and validates that it behaves correctly both with and without that change.

I have done this, but just on Windows. I hadn't thought about your point about mixing different newlines across environments, thank you! I'll change that

Reply:
@drewnoakes these are the args I used to test: -arg1\r\n-arg2\narg3\narg4\r\narg5